src/VirtualStruct.cpp

```diff
@@ -502,21 +538,58 @@ bool VirtualStruct::applyAction(const etiss::fault::Fault &fault, const etiss::f
             errormsg = std::string("No such field: ") + getInjectorPath() + "." + action.getTargetField();
             return false;
         }
-        if (f->flags_ & Field::A)
+        if (f->flags_ & (Field::A | Field::F))
         {
             return f->applyAction(fault, action, errormsg);
         }
-        else if (f->flags_ & Field::F)
-        {
-            return f->applyBitflip(action.getTargetBit(), fault.id_);
-        }
     }
         return false;
     default:
         return false;
     }
 }
 
+bool VirtualStruct::update_field_access_rights(const std::string& field, etiss::fault::Action::Type type, std::string &errormsg)
+{
+    Field* f = nullptr;
+    auto find = fieldNames_.find(field);
+    if (find == fieldNames_.end())
+    {
+        find = fieldPrettyNames_.find(field);
+        if (find == fieldPrettyNames_.end())
+        {
+            f = 0;
```

Comment:
is already set

Reply:
not sure if that fix is correct. the line you removed was searching a different list. what i meant is that f is already initialized to null at the start of the function, so this assignment is redundant and the conditional block can be dropped.

Reply:
you are right. reversed and fixed with https://github.com/tum-ei-eda/etiss/pull/100/commits/c86cb7ff136031e484abd2fb8aad1e37c39c205c